ddtrace/span.py

```diff
@@ -116,6 +112,20 @@ def start(self):
     def start(self, value):
         self.start_ns = int(value * 1e9)
 
+    @property
+    def finished(self):
+        return self.duration_ns is not None
+
+    @finished.setter
+    def finished(self, value):
+        if value:
+            if not self.finished:
+                self.duration_ns = time_ns() - self.start_ns
+            else:
+                log.warning("Ignoring finish on already finished span.")
```

Comment:
Please don't log and hide a programming error. It makes it extremely hard to debug or even write tests.

Reply:
Hmm true. I think we could do away with it altogether and document that any subsequent `finish = True` will be ignored (the same logic as for .finish()`)

Reply:
Works for me. As long as you document it, I think it's 100% fine.